File: backend/schemas/campaign.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class ProjectCreate(BaseModel):
    name: str
    campaign_objective: str  # OUTCOME_SALES, OUTCOME_TRAFFIC, etc.
    budget_type: str = "CBO"
    daily_budget: Optional[float] = None
    lifetime_budget: Optional[float] = None
    ad_account_id: str
    destination_url: Optional[str] = None


class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    campaign_objective: Optional[str] = None
    budget_type: Optional[str] = None
    daily_budget: Optional[float] = None
    lifetime_budget: Optional[float] = None
    drive_folder_url: Optional[str] = None
    naming_template: Optional[str] = None
    destination_url: Optional[str] = None


class AdSetCreate(BaseModel):
    name: str
    targeting_json: Optional[str] = None
    placements_json: Optional[str] = None
    budget: Optional[float] = None
    bid_strategy: str = "LOWEST_COST_WITHOUT_CAP"
    optimization_goal: str = "OFFSITE_CONVERSIONS"


class AdCreate(BaseModel):
    name: str
    creative_ref: Optional[str] = None
    headline: Optional[str] = None
    primary_text: Optional[str] = None
    description: Optional[str] = None
    cta: str = "SHOP_NOW"
    url: Optional[str] = None


class AdResponse(BaseModel):
    id: int
    name: str
    generated_name: Optional[str] = None
    creative_ref: Optional[str] = None
    headline: Optional[str] = None
    primary_text: Optional[str] = None
    description: Optional[str] = None
    cta: str
    url: Optional[str] = None
    url_tags: Optional[str] = None
    meta_ad_id: Optional[str] = None
    status: str

    model_config = {"from_attributes": True}


class AdSetResponse(BaseModel):
    id: int
    name: str
    generated_name: Optional[str] = None
    targeting_json: Optional[str] = None
    placements_json: Optional[str] = None
    budget: Optional[float] = None
    bid_strategy: str
    optimization_goal: str
    meta_adset_id: Optional[str] = None
    status: str
    ads: List[AdResponse] = []

    model_config = {"from_attributes": True}


class ProjectResponse(BaseModel):
    id: int
    name: str
    status: str
    campaign_objective: Optional[str] = None
    budget_type: str
    daily_budget: Optional[float] = None
    lifetime_budget: Optional[float] = None
    ad_account_id: Optional[str] = None
    drive_folder_url: Optional[str] = None
    meta_campaign_id: Optional[str] = None
    destination_url: Optional[str] = None
    created_at: Optional[datetime] = None
    ad_sets: List[AdSetResponse] = []

    model_config = {"from_attributes": True}

File: frontend/src/types/campaign.ts
export interface Project {
  id: number
  name: string
  status: string
  campaign_objective: string | null
  budget_type: string
  daily_budget: number | null
  lifetime_budget: number | null
  ad_account_id: string | null
  drive_folder_url: string | null
  meta_campaign_id: string | null
  destination_url: string | null
  created_at: string | null
  ad_sets: AdSet[]
}

export interface AdSet {
  id: number
  name: string
  generated_name: string | null
  targeting_json: string | null
  placements_json: string | null
  budget: number | null
  bid_strategy: string
  optimization_goal: string
  meta_adset_id: string | null
  status: string
  ads: Ad[]
}

export interface Ad {
  id: number
  name: string
  generated_name: string | null
  creative_ref: string | null
  headline: string | null
  primary_text: string | null
  description: string | null
  cta: string
  url: string | null
  url_tags: string | null
  meta_ad_id: string | null
  status: string
}

export interface AuthStatus {
  meta_connected: boolean
  meta_ad_account_id: string | null
  meta_page_id: string | null
  meta_business_id: string | null
  meta_business_name: string | null
  google_connected: boolean
  // Aliases for convenience
  business_name?: string | null
  business_id?: string | null
}

export interface MetaPage {
  id: string
  name: string
}

export interface Creative {
  id: number
  original_name: string
  base_name: string | null
  format: string | null
  aspect_ratio: string | null
  media_type: string | null
  file_size_bytes: number | null
  upload_status: string
}

export interface DeployEvent {
  step: string
  entity: string
  status: string
  detail?: string
  success?: boolean
  campaign_id?: string
  errors?: string[]
}

export interface CreativeAnalysis {
  creative_id: number
  filename: string
  media_type: string
  analysis: {
    creative_analysis: {
      message: string
      tone: string
      target_audience: string
      awareness_level: string
      angle: string
    }
    recommendation: {
      action: string
      adset_id: number | null
      adset_name: string
      confidence: number
      reasoning: string
      new_adset_suggestion?: {
        name: string
        description: string
        angle: string
      }
    }
  }
}

export interface BusinessPortfolio {
  id: string
  name: string
}

export interface BusinessAdAccount {
  id: string
  name: string
  account_status: number
  currency: string
}

export interface MetaCampaign {
  id: string
  name: string
  status: string
  objective: string | null
  daily_budget: number | null
}

export interface CreativeWithThumbnail extends Creative {
  thumbnail_url: string | null
  selected: boolean
  adset_name: string | null
}

export interface CreativeAssignment {
  mode: 'subfolder' | 'flat'
  assignments: Record<string, CreativeWithThumbnail[]>
}

export interface ParseResult {
  status: string
  structure: any
  missing_fields: string[]
  ad_sets_count: number
  total_ads: number
}
